.github/workflows/lint.yml

```diff
@@ -0,0 +1,50 @@
+# lint runs all linters in this repository
+# This workflow is triggered by ci_release.yml workflow
+name: lint
+on:
+  workflow_call:
+
+jobs:
+  golangci-lint:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: ./go.mod
+        # This steps sets the GIT_DIFF environment variable to true
+        # if files defined in PATTERS changed
+      - uses: technote-space/get-diff-action@v6.1.2
+        with:
+          # This job will pass without running if go.mod, go.sum, and *.go
+          # wasn't modified.
+          PATTERNS: |
+            **/**.go
+            go.mod
+            go.sum
+      - uses: golangci/golangci-lint-action@v6.1.0
+        with:
+          version: latest
+          args: --timeout 10m
+          github-token: ${{ secrets.github_token }}
+        if: env.GIT_DIFF
+
+  # hadolint lints the Dockerfile
```

Comment:
Same as the dependabot, if we don't plan on having a dockerfile, i'd just delete

Reply:
Created #10. For now I'll leave this commented out.